### PR TITLE
Switch AES C implementation to PQClean's

### LINKED ISSUE #949: Add constant-time plain C AES implementation when building without AES-NI
There's such an implementation available in PQClean that we could use.

### PR DESCRIPTION
**Pull request by dstebila**
Fixes #949. 

* [No] Does this PR change the input/output behaviour of a cryptographic algorithm (i.e., does it change known answer test values)?  (If so, a version bump will be required from *x.y.z* to *x.(y+1).0*.)
* [No] Does this PR change the the list of algorithms available -- either adding, removing, or renaming?  (If so, PRs in OQS-OpenSSL, OQS-BoringSSL, and OQS-OpenSSH will also be required by the time this is merged.)

### CODE REVIEW

File: src/common/aes/aes_local.h [hunk 1]
```diff
@@ -9,15 +9,15 @@ void oqs_aes128_ecb_enc_sch_ni(const uint8_t *plaintext, const size_t plaintext_
 
 void oqs_aes128_load_schedule_c(const uint8_t *key, void **_schedule);
 void oqs_aes128_free_schedule_c(void *schedule);
-void oqs_aes128_enc_sch_block_c(const uint8_t *plaintext, const void *_schedule, uint8_t *ciphertext);
 void oqs_aes128_ecb_enc_sch_c(const uint8_t *plaintext, const size_t plaintext_len, const void *schedule, uint8_t *ciphertext);
```
Comment (jschanck) [inline src/common/aes/aes_local.h:12]: Remove the `const` from `const size_t plaintext_len`.

File: src/common/aes/aes_local.h [hunk 2]
```diff
@@ -9,15 +9,15 @@ void oqs_aes128_ecb_enc_sch_ni(const uint8_t *plaintext, const size_t plaintext_
...
 
 void oqs_aes256_load_schedule_ni(const uint8_t *key, void **_schedule);
 void oqs_aes256_free_schedule_ni(void *schedule);
 void oqs_aes256_enc_sch_block_ni(const uint8_t *plaintext, const void *_schedule, uint8_t *ciphertext);
 void oqs_aes256_ecb_enc_sch_ni(const uint8_t *plaintext, const size_t plaintext_len, const void *schedule, uint8_t *ciphertext);
+void oqs_aes256_ctr_enc_sch_ni(const uint8_t *iv, const size_t iv_len, const void *schedule, uint8_t *out, size_t out_len);
```
Comment (jschanck) [inline src/common/aes/aes_local.h:18]: Remove `const` from `const size_t iv_len`.

**Comment by jschanck** [comment]
Not technically related to this PR, but I also noticed some decrypt functions still in aes128_ni.c and aes256_ni.c.

File: src/common/aes/aes.c [hunk 3]
```diff
@@ -129,7 +116,7 @@ static uint32_t UINT32_TO_BE(const uint32_t x) {
 }
 #define BE_TO_UINT32(n) (uint32_t)((((uint8_t *) &(n))[0] << 24) | (((uint8_t *) &(n))[1] << 16) | (((uint8_t *) &(n))[2] << 8) | (((uint8_t *) &(n))[3] << 0))
 
-void OQS_AES256_CTR_sch(const uint8_t *iv, size_t iv_len, const void *schedule, uint8_t *out, size_t out_len) {
+void oqs_aes256_ctr_enc_sch_ni(const uint8_t *iv, const size_t iv_len, const void *schedule, uint8_t *out, size_t out_len) {
```
[the remark this answers is not in the document]
Comment (dstebila) [reply src/common/aes/aes.c:101]: Done.

File: src/common/aes/aes_c.c [hunk 4]
```diff
@@ -1,273 +1,647 @@
...
+	size_t i;
+	uint32_t cc;
+
+	if (iv_len == 12) {
+		br_range_dec32le(ivw, 3, iv);
+		ivw[3] = 0;
+	} else if (iv_len == 16) {
+		br_range_dec32le(ivw, 4, iv);
+	} else {
+		assert(1 == 0);
```
[the remark this answers is not in the document]
Comment (dstebila) [reply src/common/aes/aes_c.c]: Done.

File: src/common/aes/aes_local.h (hunk 1 above)
Comment (dstebila) [reply src/common/aes/aes_local.h:12]: We have const size_t in a bunch of other places, which is technically unnecessary but I'll keep for now for consistency.

**Review by jschanck — APPROVED** [review]

**Pull request merged**